src/lib/data/alignment-group.js

```diff
@@ -0,0 +1,43 @@
+import { v4 as uuidv4 } from 'uuid'
+
+export default class AlignmentGroup {
+  constructor (token) {
+    this.id = uuidv4()
+    this.origin = []
+    this.target = []
+    this.steps = []
+    this.add(token)
+  }
+
+  add (token) {
+    this[token.textType].push(token.idWord)
+    this.steps.push({ textType: token.textType, id: token.idWord })
+  }
+
+  get lastStepTextType () {
+    return this.steps[this.steps.length - 1].textType
+  }
+
+  get couldBeFinished () {
+    return this.id && this.origin && this.origin.length > 0 && this.target && this.target.length > 0
+  }
+
+  couldBeAdded (token) {
+    return !this.couldBeFinished || (this.lastStepTextType === token.textType)

```

Comment:
I find the check on couldBeFinished to be a little confusing here. Is it really the same as checking whether lastStepTextType is set at all?

Reply:
yes, you are right - it is unuseful method in the updated start/end alignmentGroup workflow
I removed it

Reply:
I don't see a chance in this code in the latest commit. I'm fine with merging without it if you add an issue to come back to it though.

Reply:
it is strange, I believe, I forgot to do a commit - will repeat this change